packages/accounts-controller/src/AccountsController.ts

```diff
@@ -423,6 +429,10 @@ export class AccountsController extends BaseController<
     }
 
     this.updateAccountMetadata(accountId, { name: accountName });
+
+    const account = this.getAccountExpect(accountId);
+
+    this.messagingSystem.publish('AccountsController:accountRenamed', account);
```

Comment:
If the caller uses `updateAccountMetadata` instead to update multiple metadata at the same time, we won't fire this event 😬 (and since our current document is:

```text
Use {@link setAccountName} if you only need to update the name of the account.
```

But IIRC, we agreed with this pattern, the caller needs to know this and always call the `setAccountName` to update the name.

We can talk internally about this!

Reply:
You're definitely right, updating this!

Reply:
After discussing internally, here's the plan:

1. We keep the event firing in the `setAccountName`
2. We refactor (in another PR) to locate name uniqueness check + event firing in the `updateMetadata` method, thus making sure all "paths" are covered (the caller can either call `updateMetadata` or `setAccountName` and all validations + events will be handled)

Meaning, `setAccountName` will become something like:
```ts
setAccountName(name: string) {
  // This will check for name uniqueness and fire `accountRenamed` if the account
  // has been renamed
  this.updateMetadata({ name });
}
```